kolena/_experimental/object_detection/utils.py

```diff
@@ -177,3 +177,96 @@ def compute_confusion_matrix_plot(
     for actual_label in ordered_labels:
         matrix.append([confusion_matrix[actual_label][predicted_label] for predicted_label in ordered_labels])
     return ConfusionMatrix(title=plot_title, labels=ordered_labels, matrix=matrix)
+
+
+def _compute_sklearn_arrays_by_class(
+    all_matches: List[MulticlassInferenceMatches],
+) -> Tuple[Dict[str, np.ndarray], Dict[str, np.ndarray]]:
+    y_true_by_label: Dict[str, np.ndarray] = {}
+    y_score_by_label: Dict[str, np.ndarray] = {}
+
+    labels: Set[str] = set()
+    for match in all_matches:
+        for _, bbox_inf in match.matched:
+            labels.add(bbox_inf.label)
+        for bbox_gt, _ in match.unmatched_gt:
+            labels.add(bbox_gt.label)
+        for bbox_inf in match.unmatched_inf:
+            labels.add(bbox_inf.label)
+
+    for label in labels:
+        filtered_matchings = [
+            InferenceMatches(
+                matched=[(gt, inf) for gt, inf in match.matched if gt.label == label],
+                unmatched_gt=[gt for gt, _ in match.unmatched_gt if gt.label == label],
+                unmatched_inf=[inf for inf in match.unmatched_inf if inf.label == label],
+            )
+            for match in all_matches
+        ]
+
+        y_true, y_score = _compute_sklearn_arrays(filtered_matchings)
+        y_true_by_label[label] = y_true
+        y_score_by_label[label] = y_score
+
+    return y_true_by_label, y_score_by_label
+
+
+def _compute_optimal_f1_with_arrays(
+    y_true: np.ndarray,
+    y_score: np.ndarray,
+) -> float:
+    # optimal threshold is 0 if there is no relevant or true label
+    if np.all(y_true == 0):
+        return 0.0
+
+    precision, recall, thresholds = sklearn_metrics.precision_recall_curve(y_true, y_score)
```

Comment:
I believe if `y_true` and `y_score` is both empty then this call will fail.  Can we add a check at the top and add a unit test for this case?

Reply:
thanks for catching this 👍